doc/source/whatsnew/v1.3.0.rst

```diff
@@ -932,6 +932,7 @@ Timezones
 ^^^^^^^^^
 - Bug in different ``tzinfo`` objects representing UTC not being treated as equivalent (:issue:`39216`)
 - Bug in ``dateutil.tz.gettz("UTC")`` not being recognized as equivalent to other UTC-representing tzinfos (:issue:`39276`)
+- Bug in :func:`to_datetime` with ``infer_datetime_format=True`` failing to parse zero UTC offset (``Z``) correctly (:issue:`41047`)
```

Comment:
Sorry for the delay. This PR didn't make it into 1.3. Could you move this to 1.4.0 whatsnew?

Reply:
@mroeschke OK!
Should I merge the current master and then fix the whatsnew file?

Reply:
Yes that would be great, thank you.

Reply:
Fixed whatsnew entry.